yetanotherco/aligned_layer
# feat(audit): Emit events in Operator whitelist

Closes #924 

-

## To Test
- Verify you are on the submodule branch
```
cd contracts/lib/eigenlayer-middleware
git status
```
You should be on the `924-emit-events-in-whitelist-add-remove` branch.

- Deploy the contracts before running a local testnet by first running `make anvil_deploy_eigen_contracts`, then `make anvil_deploy_aligned_contracts`, then setting up a running testnet following the testnet guide.
- Run the whitelist operator script via `make operator_whitelist_devnet` and remove it via `make operator_remove_devnet`.
- Use `cast` to check the events have been emitted
```
cast logs --from-block 0 --to-block latest "AddedToWhiteList(address indexed added)" --rpc-url http://localhost:8545 --address 0x851356ae760d987E095750cCeb3bC6014560891C

cast logs --from-block 0 --to-block latest "RemovedFromWhiteList(address indexed added)" --rpc-url http://localhost:8545 --address 0x851356ae760d987E095750cCeb3bC6014560891C
```
